src/main/webapp/data-visualization.js

```diff
@@ -508,6 +529,142 @@ function toggleDataTable() {
   }
 }
 
+function displayYearlyChart(chartId, data, description, location) {
+  const chart = am4core.create(chartId, am4charts.XYChart);
+  chart.data = data;
+  const dateAxis = chart.xAxes.push(new am4charts.DateAxis());
+  dateAxis.dataFields.category = 'year';
+  dateAxis.title.text = 'Year';
+  dateAxis.baseInterval = {timeUnit: 'year'};
+  const valueAxis = chart.yAxes.push(new am4charts.ValueAxis());
+  valueAxis.title.text = description;
+  const series = chart.series.push(new am4charts.LineSeries());
+  series.dataFields.valueY = 'value';
+  series.dataFields.dateX = 'year';
+  series.name = description;
+  series.tooltip.getFillFromObject = true;
+  series.tooltipText = '{valueY}';
+  series.strokeWidth = 3;
+
+  const color = getColor();
+  series.stroke = am4core.color(color);
+  series.yAxis = valueAxis;
+
+  const bullet = series.bullets.push(new am4charts.CircleBullet());
+  bullet.stroke = am4core.color('#fff');
+  bullet.fill = am4core.color(color);
+  bullet.strokeWidth = 2;
+  bullet.tooltipText = '{valueY}';
+  bullet.circle.radius = 4;
+
+  const mapTitle = chart.titles.create();
+  let title;
+  const state = stateInfo[location];
+  if (state != null) {
+    title = `${description} in ${state.name} by year`;
+  } else {
+    title = `${description} in ${location} by year`;
+  }
+  mapTitle.text = title;
+
+  chart.legend = new am4charts.Legend();
+  document.getElementById(chartId).style.display = 'block';
+}
+
+// Create yearly state total population growth chart for the submitted query
+async function createYearlyStateTotalChart(
+  personType, action, location, description) {
+  document.getElementById('county-form').style.display = 'block';
+  const loadingMsg = document.getElementById('yearly-total-chart-loading-msg');
+  loadingMsg.innerText = 'Yearly state data chart loading...';
+  const data = [];
+  for (let year = 2010; year <= 2018; year++) {
+    await fetch(getFetchUrl('query', personType, action, location, year) +
+      '&state-total=true')
+      .then((response) => response.json().then((jsonResponse) => ({
+        data: jsonResponse,
+        success: response.ok,
+        status: response.status,
+      })))
+      .then((response) => {
+        if (response.success) {
```

Comment:
If response isn't success, you should display an error

Reply:
Right now I show a "no data available" message if there is no data returned successfully for 2010-2018. If I add an error message on not success, it'd keep showing for all of the years, but also show the graph with data for the successful years. So, I was thinking the "no data available" message when all years are not success would be the most useful to the user. Let me know what you'd had in mind for error messages

Reply:
I see that you do that for the individual county graphs, but it looks like you don't for the state graph? I agree that no data available makes sense.

Reply:
Ah, you're right, added error message for state charts